### PR TITLE
Consider python 3.5 async statement

### PR DESCRIPTION
**Pull request by nramirezuy**
``` python
async def function():
    pass


class Foo(object):

    async def method():
        pass
```

source: [PEP-0492](https://www.python.org/dev/peps/pep-0492/#new-coroutine-declaration-syntax)

### CODE REVIEW

File: autoload/braceless.vim [hunk 1]
```diff
@@ -11,7 +11,7 @@ let s:pattern_cache = {}
 " Default patterns
 let s:pattern_default = {}
 let s:pattern_default.python = {
-      \   'start': '\<\%(if\|def\|for\|try\|elif\|else\|with\|class\|while\|except\|finally\)\>\_.\{-}:\ze\s*\%(\_$\|#\)',
+      \   'start': '\<\%(async\|if\|def\|for\|try\|elif\|else\|with\|class\|while\|except\|finally\)\>\_.\{-}:\ze\s*\%(\_$\|#\)',
```
Comment (tweekmonster) [inline autoload/braceless.vim:14]: If there's no other place for `async` to appear, it might be better if it's in the pattern as `async def`.

**Comment by tweekmonster** [comment]
Would you mind adding some tests for this?

**Comment by nramirezuy** [comment]
@tweekmonster My Docker install broke, so I can't run tests. :cry:

I think it will be better if I close this PR and you just make it happen. :smile:

**Comment by tweekmonster** [comment]
@nramirezuy You could always just let Travis-CI verify the tests.  It'd be just copying some of the tests that have `def` in them, but making sure `async def` is treated separately.

In hindsight, I believe there's a few special cases for `def` that might need to include `async def` as well.  I don't have enough time to do this now, but feel free to reopen if you have the time.

**Comment by blueyed** [comment]
Yeah @nramirezuy - please contribute to things you love/use.

We're happy to help, but have limited resources ourselves.

The PR/issue is valid by itself, so it should not be closed.

**Pull request closed without merging**